orttraining/orttraining/core/graph/gradient_builder.cc

```diff
@@ -1794,7 +1794,18 @@ IMPLEMENT_GRADIENT_BUILDER(GetExternalGradient) {
     }
 
     std::vector<ArgDef> output_args;
-    for (const auto& output : node_def.outputs) {
+    for (size_t output_index = 0; output_index < node_def.outputs.size(); ++output_index) {
+      if (output_index >= GetSrcNodeInputSize()) {
```

Comment:
Maybe it is worth adding a comment about a case it where this condition is needed.

Reply:
@xadupre I've included your suggestion in https://github.com/microsoft/onnxruntime/pull/21107